File: Napakalaki/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.Random;

/**
 *
 * @author carlos
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers=0;
    
    Cultist myCultistCard;
    
    CultistPlayer(Player p, Cultist c){
        super(p);
        
        totalCultistPlayers++;
        this.myCultistCard=c;
    }
    
    @Override
    protected int getCombatLevel(){
        int final_level=super.getCombatLevel()+ ((super.getCombatLevel())*20/100);
        final_level= final_level + (this.myCultistCard.getGainedLevels() * totalCultistPlayers);
        
        return final_level;
    }
    
    protected int getOponentLevel(Monster m){
       return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    private Treasure giveMeATreasure(){
        Random r = new Random();
        return visibleTreasures.get(r.nextInt(hiddenTreasures.size()));
    }
    
    private boolean canYouGiveMeATreasure(){
        if(!visibleTreasures.isEmpty())
            return true;
        else
            return false;
    }
    
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}

File: Napakalaki-Ruby/lib/Cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class CultistPlayer < Player
  
  #Atributos
  attr_accessor :myCultistCard
  
  @@totalCultistPlayers=0
  
  def initialize(player,c)
    super(player.name)
    
    ConstructorCopia(player)
    
    @myCultistCard = c
    @@total_cultist_players += 1
  end
  
  
  #protected
  def getCombatLevel
      final_level=super.getCombatLevel + ((super.getCombatLevel)*20/100)
      final_level= final_level + (@myCultistCard.getGainedLevels * @@totalCultistPlayers);
        
      return final_level
  end
    
  def getOponentLevel(m)
     return m.getCombatLevelAgainstCultistPlayer
  end
    
  def shouldConvert()
        return false
  end
    
  def Treasure giveMeATreasure
    return @visibleTreasures.get(rand(hiddenTreasures.length));
  end
    
  def canYouGiveMeATreasure
    if(!@visibleTreasures.empty?)
      return true
    else
      return false
    end
  end
    
  def self.getTotalCultistPlayers
    return @@totalCultistPlayers
  end
end
